modules/siddhi-extensions/event-table/src/main/java/org/wso2/siddhi/extension/eventtable/hazelcast/HazelcastOperatorParser.java

```diff
@@ -0,0 +1,215 @@
+     *
+     * @param metaComplexEvent    Meta information about ComplexEvent.
+     * @param matchingStreamIndex Matching stream index
+     * @param expression          Expression
+     * @param candidateDefinition candidate definition
+     * @param indexedAttribute    Indexed attribute
+     * @return boolean representing whether an expression is a indexed variable
+     */
+    private static boolean isTableIndexVariable(MetaComplexEvent metaComplexEvent, int matchingStreamIndex, Expression expression,
+                                                AbstractDefinition candidateDefinition, String indexedAttribute) {
+        if (expression instanceof Variable) {
+            Variable variable = (Variable) expression;
+            if (variable.getStreamId() != null) {
+                MetaStreamEvent tableStreamEvent = getTableMetaStreamEvent(metaComplexEvent, matchingStreamIndex, candidateDefinition);
+                if (tableStreamEvent != null) {
+                    if ((tableStreamEvent.getInputReferenceId() != null && variable.getStreamId().equals(tableStreamEvent.getInputReferenceId())) ||
+                            (tableStreamEvent.getLastInputDefinition().getId().equals(variable.getStreamId()))) {
+                        if (Arrays.asList(tableStreamEvent.getLastInputDefinition().getAttributeNameArray()).contains(indexedAttribute)) {
+                            return true;
+                        }
+                    }
+                } else {
+                    if (candidateDefinition.getId().equals(variable.getStreamId())) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the table meta stream event
+     *
+     * @param metaComplexEvent    Meta information about ComplexEvent.
+     * @param matchingStreamIndex Matching stream index
+     * @param candidateDefinition candidate definition
+     * @return Meta stream event
+     */
+    private static MetaStreamEvent getTableMetaStreamEvent(MetaComplexEvent metaComplexEvent, int matchingStreamIndex,
+                                                           AbstractDefinition candidateDefinition) {
+        if (metaComplexEvent instanceof MetaStateEvent) {
+            MetaStreamEvent[] metaStreamEvents = ((MetaStateEvent) metaComplexEvent).getMetaStreamEvents();
+            int candidateEventPosition = 0;
+            //for join
+            for (; candidateEventPosition < metaStreamEvents.length; candidateEventPosition++) {
+                MetaStreamEvent metaStreamEvent = metaStreamEvents[candidateEventPosition];
+                if (candidateEventPosition != matchingStreamIndex && metaStreamEvent.getLastInputDefinition().equalsIgnoreAnnotations(candidateDefinition)) {
+                    return metaStreamEvent;
+                }
+            }
+        }
+        return null;
+    }
+}
```

Comment:
There are several places in class where you exceed line width of 200. Please adhere to WSO2 coding styles